src/buffer.rs

```diff
@@ -479,6 +682,18 @@ where
         }
     }
 
+    /// Enumerates over the rows of the image.
+    /// The iterator yields the y-coordinate of each row
+    /// along with a mutable reference to them.
+    pub fn enumerate_rows_mut(&mut self) -> EnumerateRowsMut<P> {
+        let width = self.width;
+        EnumerateRowsMut {
+            rows: self.rows_mut(),
+            y: 0,
+            width,
+        }
+    }
+
```

Comment:
Can this get some tests? E.g. the multiplication bug can be tested not to occur with an RGB image of size (2^31, 1).

Reply:
Hmm... Won't this work only on 64-bit architectures, though, so that the test would have to only be run on such? And won't such test allocate multiple GBs of memory (maybe that's not a problem, though)?

Reply:
Agh, you are correct. This could be faked though with a test-only implementation of `Pixel` and `Subpixel` that are both zero-sized types, thus not requiring any allocation in their `Vec<_>` etc. The image itself would be just small enough to also not panic on 32-bit (I think?) and there not do anything useful but also not fail?

But overall, yeah, this is pretty overkill not worth it. I initially thought it was easier.